session.go

```diff
@@ -262,6 +262,9 @@ func (s *Session) RunAnalyzeQuery(ctx context.Context, stmt spanner.Statement) (
 
 func (s *Session) runQueryWithOptions(ctx context.Context, stmt spanner.Statement, opts spanner.QueryOptions) (*spanner.RowIterator, *spanner.ReadOnlyTransaction) {
 	if s.InReadWriteTransaction() {
+		// The current Go Spanner client library does not apply client-level directed read options to read-write transactions.
+		// Therefore, we explicitly set query-level options here to fail the query during a read-write transaction.
+		opts.DirectedReadOptions = s.clientConfig.DirectedReadOptions
```

Comment:
The current Go Spanner client library does not seem to apply client-level directed read options to RW Tx. Therefore, even if we set the directed reads option, queries executed within a RW Tx will be routed to the read-write replicas of the leader region.

From the perspective of spanner-cli usage, it may be preferable for queries to fail when a RW Tx is performed with directed read options set. So, we explicitly set query-level options to cause queries to fail during RW Tx with the following error:

```
spanner(rw txn)> select 1;
ERROR: spanner: code = "InvalidArgument", desc = "Directed reads can only be performed in a read-only transaction."
```